src/App.js

```diff
@@ -1,16 +1,34 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const totalLikes = messages.reduce((total, entry) => {
+    total += entry.liked;
+    return total;
+  }, 0);
```

Comment:
👍 Nice use of `reduce` to tally up the total number of liked messages, avoiding the introduction of an additional piece of state. This approach does need to iterate over the message list. However, since we map over the list later anyway, the additional cost of the iteration here is negligible. If we did end up with other interactions that caused a re-render and didn't want to perform the calculation unnecessarily, there are ways to memoize results in React that can be used to avoid unnecessary calculations. But we should start by deriving values from existing state when possible, and only look for optimizations if the rendering performance is adversely impacted.